discovery/consul/consul_test.go

```diff
@@ -297,3 +297,89 @@ func TestAllOptions(t *testing.T) {
 	checkOneTarget(t, <-ch)
 	cancel()
 }
+
+func TestGetDatacenter_ShouldUseAgentInfo(t *testing.T) {
```

Comment:
This path should already be covered by the existing tests AFAICT.

Reply:
As said above, it should already be covered so I'd rather remove this test.

Reply:
Test was removed.